File: references/cellab_devkit-1.0 (1)/ruledefs/rug.js
/*

    This program runs an eight cell averaging rule of eight bits
    per cell.  We program it as a nowrap worldtype 12 calling
    the semi8 evaluator.

*/

    rule.worldtype = 12;         // User evaluator, 2D open world
    rule.ocodereq = "semi8";

    function rug(lutindex) {
        return ((lutindex >> 3) + 1) & 0xFF;
    }

File: references/cellab_devkit-1.0 (1)/rules_java/rug.java
/*

    This program runs an eightcell averaging rule of eight bits per
    cell.  We program it as a nowrap owncode WorldType 12 calling
    semi8.jco.

*/

class Rug extends ruletable {
    void jcruleModes() {
        setWorld(12);
        setOwnCodeRequest("semi8");
    }

    int jcrule(int oldstate) {
        return ((oldstate >> 3) + 1) & 0xFF;
    }
}

public class rug {
    public static void main(String args[]) {
        (new Rug()).generateRuleFile("rug");
    }
}
